v2/delete.go

```diff
@@ -0,0 +1,10 @@
+package pie
+
+// Removes element at index idx from input slice, returns resulting slice.
+// If the index out of bounds, returns unchanged input slice.
+func Delete[T any](ss []T, idx int) []T {
```

Comment:
This also needs to be registered against the chainable types: https://github.com/elliotchance/pie/blob/master/v2/of.go, https://github.com/elliotchance/pie/blob/master/v2/of_numeric.go and https://github.com/elliotchance/pie/blob/master/v2/of_ordered.go

Reply:
It also makes sense to me to allow zero or more indexes:

```
.Delete(3, 4, 5)
.Delete(bunchOfIndexes...)
```

Although this is a little more complicated, so I'm happy for that to come later if needed.